scipy/optimize/_nnls.py

```diff
@@ -72,11 +72,13 @@ def nnls(A, b, *, maxiter=None, atol=_NoValue):
     b = np.asarray_chkfinite(b, dtype=np.float64)
 
     if len(A.shape) != 2:
-        raise ValueError("Expected a two-dimensional array (matrix)" +
-                         f", but the shape of A is {A.shape}")
-    if len(b.shape) != 1:
-        raise ValueError("Expected a one-dimensional array (vector)" +
-                         f", but the shape of b is {b.shape}")
+        raise ValueError(f"Expected a 2D array, but the shape of A is {A.shape}")
+
+    if (b.ndim > 2) or ((b.ndim == 2) and (b.shape[1] != 1)):
+        raise ValueError("Expected a 1D array,(or 2D with one column), but the,"
+                         f" shape of b is {b.shape}")
```

Comment:
Before , `b` was allowed to be only 1D, this would allow also `(n, 1)`. Why should we change this?

Reply:
Just consistency in terms of problem definition. Since we do this everywhere else, I did not think of any reason to reject it because `Ax = b` is consistent for such `b`. Sometimes it might be coming from another piece of code that slices a column and it's just these extra unnecessary disruptions that annoy users the most. I don't have strong opinions though either way is fine for me.

Reply:
Ah, in `linalg`, this shape is typically allowed? Then I agree, let's do so also for `nnls`. Just let's leave it for a followup and add tests if that is alright for you?